tests/unit_tests/dao/base_dao_test.py

```diff
@@ -37,6 +79,1924 @@ class TestDAOWithNoneModel(BaseDAO[MockModel]):
+    """Test that model_cls property is properly set."""
+    assert UserDAO.model_cls == User
+
+    # Test with a DAO that doesn't set model_cls
+    class InvalidDAO(BaseDAO[User]):
+        pass
+
+    # Should have model_cls attribute set by __init_subclass__
+    # Just verify it exists and is the expected type
+    assert hasattr(InvalidDAO, "model_cls")
+    assert InvalidDAO.model_cls == User  # It inherits the generic type
+
+
+def test_base_dao_id_column_name_property(app_context: None) -> None:
+    """Test that id_column_name property is properly used."""
+    assert UserDAO.id_column_name == "id"
+
+    # Test custom id column name
+    class CustomIdDAO(BaseDAO[User]):
+        model_cls = User
+        id_column_name = "username"  # Use username as ID column
+
+    # Create test user
+    user = User(
+        id=5016,
+        username="unique-username-id",
+        first_name="Custom",
+        last_name="User",
+        email="custom@example.com",
+        active=True,
+    )
+    db.session.add(user)
+    db.session.commit()
+
+    # Find by custom ID column
+    found = CustomIdDAO.find_by_id("unique-username-id")
+    assert found is not None
+    assert found.first_name == "Custom"
+
+    # Cleanup
+    db.session.delete(user)
+    db.session.commit()
+
+
+def test_base_dao_base_filter_integration(app_context: None) -> None:
+    """Test base_filter integration with list operations."""
+    # Create mix of users
+    users = []
+    for i, active in enumerate([True, False, True, False, True]):
+        user = User(
+            id=2100 + i,
+            username=f"base_filter_user_{i}",
+            first_name="BaseFilter",
+            last_name=f"User{i}",
+            email=f"basefilter{i}@example.com",
+            active=active,
+        )
+        users.append(user)
+        db.session.add(user)
+    db.session.commit()
+
+    # Test basic list functionality
+    results, total = UserDAO.list()
+    assert isinstance(results, list)
+    assert isinstance(total, int)
+    assert total >= 5  # Should include our 5 test users
+
+    # Test skip_base_filter functionality
+    results_skip, total_skip = UserDAO.list()
+    assert isinstance(results_skip, list)
+    assert isinstance(total_skip, int)
+
+    # Test count functionality
+    count_all = UserDAO.count()
+    assert count_all >= 5
+
+    count_skip = UserDAO.count(skip_base_filter=True)
+    assert count_skip >= 5
+
+    # Cleanup
+    for user in users:
+        db.session.delete(user)
+    db.session.commit()
+
+
+def test_base_dao_edge_cases(app_context: None) -> None:
+    """Test BaseDAO edge cases and error conditions."""
+    # Test create without item or attributes
+    created = UserDAO.create()
+    assert created is not None
+    # User model has required fields, so we expect them to be None
+    assert created.username is None
+
+    # Don't commit - would fail due to constraints
+    db.session.rollback()
```

Comment:
I think technically you might have integration tests in here (seem to use the database, didn't analyze to see if there are module-level mocks setup). I don't think we've been great at segmenting properly, so there are probably integration tests in unit tests in Superset and vice-versa. So in theory might belong in `tests/unit_tests/` (?)

Reply:
yeah i think you're right. some of these are integration tests, and should be in `tests/integration_tests` as such. I will refactor.

Reply:
Crunching...